### DISCUSSION TITLE
Execute the same `.feature` file with different parameters? Seeking for a `Background Outline`-like functionality

### DISCUSSION THREAD

**Discussion by KonstantinosArch** [reqnroll/Reqnroll#360] [General]
Hello Reqnroll Community, 👋

I am working on a testing scenario where I need to execute the same `.feature` file multiple times with different parameters. Ideally, this would work like a `Background Outline` concept, akin to `Scenario Outline`, with an `Examples` table that provides the variables. However, I understand that such a feature does not currently exist in Reqnroll.

The goal is to reduce redundancy and maintenance overhead when dealing with groups of similar scenarios. Here is the specific use case:

The variable could be used in `Background`, eg:

```gherkin
Background: Navigate to page
    Given the admin user logs in to the site
    And the user navigates to <Page> page
```
because the exact same scenarios apply to 2 or 3 different pages. So, it would be perfect if something like this existed:

```gherkin
Background Outline: Navigate to page
    Given the admin user logs in to the site
    And the user navigates to <Page> page

 Examples:
    | Page      |
    | Home      |
    | Dashboard |
    | Reports   |
```
and the tests run once for each variable in the `Background`.

Is there a way to create something akin to `Background Outline` in Reqnroll?

For example, can we parameterize an entire `.feature` file and run it multiple times with different parameters?
Are there existing mechanisms or plugins in Reqnroll that allow similar functionality?

If not, what workarounds would you recommend for achieving this?

I found something similar from Specflow [here](https://specflow.org/challenges/groups-of-similar-scenarios), but I don't think it fits my case.

I appreciate any insights, ideas, or guidance you can provide. If needed, I’m happy to contribute to or collaborate on a feature request to bring this capability to Reqnroll. Looking forward to hearing your thoughts!

**Comment by obligaron** [comment]
Would something like https://github.com/TotalTest/SpecFlow.Contrib.Variants help in your case?

We had another discussion about the plugin and a possible port to Reqnroll, see https://github.com/orgs/reqnroll/discussions/221.
